cms/djangoapps/contentstore/outlines.py

```diff
@@ -212,6 +212,70 @@ def _make_not_bubbled_up_error(seq_usage_key, seq_group_access, user_partition_i
     )
 
 
+def _make_sequence_data(sequence):
+    sequence_errors = []
+    seq_user_partition_groups, error = _make_user_partition_groups(
+        sequence.location, sequence.group_access
+    )
+    if error:
+        sequence_errors.append(error)
+
+    # Bubble up User Partition Group settings from Units if appropriate.
+    sequence_upg_from_units = _bubbled_up_groups_from_units(
+        [unit.group_access for unit in sequence.get_children()]
+    )
+    for user_partition_id, group_ids in sequence_upg_from_units.items():
+        # If there's an existing user partition ID set at the sequence
+        # level, we respect it, even if it seems nonsensical. The hack of
+        # bubbling things up from the Unit level is only done if there's
+        # no conflicting value set at the Sequence level.
+        if user_partition_id not in seq_user_partition_groups:
+            sequence_errors.append(
+                _make_bubbled_up_error(sequence.location, user_partition_id, group_ids)
+            )
+            seq_user_partition_groups[user_partition_id] = group_ids
+        else:
+            sequence_errors.append(
+                _make_not_bubbled_up_error(
+                    sequence.location, sequence.group_access, user_partition_id, group_ids
+                )
+            )
```

Comment:
I should probably have encapsulated this in a new function called by `_make_user_partition_groups` when I made it. Might be worth doing now, as split_test complicates this logic even more.